sonic_y_cable/y_cable.py

```diff
@@ -507,3 +546,655 @@ def check_if_link_is_active_for_torB(physical_port):
+        main = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset + (lane-1)*5 + 2, 1)
+        y_cable_validate_read_data(main, 1, physical_port, "tor1 cursor result")
+        result.append(c_int8(main[0]).value)
+        post1 = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset + (lane-1)*5 + 3, 1)
+        y_cable_validate_read_data(post1, 1, physical_port, "tor1 cursor result")
+        result.append(c_int8(post1[0]).value)
+        post2 = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset + (lane-1)*5 + 4, 1)
+        y_cable_validate_read_data(post2, 1, physical_port, "tor1 cursor result")
+        result.append(c_int8(post2[0]).value)
+
+    else:
+        helper_logger.log_error("platform_chassis is not loaded, failed to get TOR 1 cursor values")
+        return -1
+
+    return result
+
+
+@hook_y_cable_simulator
+def get_tor2_cursor_values(physical_port, lane):
+    """
+    This API specifically returns the cursor equalization parameters for TOR 2.
+    This includes pre one, pre two , main, post one, post two cursor values
+
+    Args:
+        physical_port:
+             an Integer, the actual physical port connected to a Y cable
+        lane:
+             an Integer, the lane on which to collect the cursor values
+                         1 -> lane 1,
+                         2 -> lane 2
+                         3 -> lane 3
+                         4 -> lane 4
+    Returns:
+        an list, with  pre one, pre two , main, post one, post two cursor values in the order
+    """
+
+    curr_offset = Y_CABLE_TOR2_CURSOR_VALUES
+
+    result = []
+
+    if platform_chassis is not None:
+        pre1 = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset + (lane-1)*5, 1)
+        y_cable_validate_read_data(pre1, 1, physical_port, "tor 2 cursor result")
+        result.append(c_int8(pre1[0]).value)
+        pre2 = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset + (lane-1)*5 + 1, 1)
+        y_cable_validate_read_data(pre2, 1, physical_port, "tor 2 cursor result")
+        result.append(c_int8(pre2[0]).value)
+        main = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset + (lane-1)*5 + 2, 1)
+        y_cable_validate_read_data(main, 1, physical_port, "tor 2 cursor result")
+        result.append(c_int8(main[0]).value)
+        post1 = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset + (lane-1)*5 + 3, 1)
+        y_cable_validate_read_data(post1, 1, physical_port, "tor 2 cursor result")
+        result.append(c_int8(post1[0]).value)
+        post2 = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset + (lane-1)*5 + 4, 1)
+        y_cable_validate_read_data(post2, 1, physical_port, "tor 2 cursor result")
+        result.append(c_int8(post2[0]).value)
+
+    else:
+        helper_logger.log_error("platform_chassis is not loaded, failed to get NIC cursor values")
+        return -1
+
+    return result
```

Comment:
This code is duplicated for NIC, ToR1 and ToR2. Suggest creating a helper function which takes the starting offset as a parameter.

Reply:
fixed